spec/providers/aws/sqs_broker_spec.rb

```diff
@@ -57,29 +57,110 @@
+          \"MessageAttributes\" : {
+            \"eventId\" : {\"Type\":\"String\",\"Value\":\"#{eventId}\"},
+            \"context\" : {\"Type\":\"String\",\"Value\":\"#{context}\"},
+            \"correlationId\" : {\"Type\":\"String\",\"Value\":\"#{correlationId}\"}
+          }
+        }
+        EOS_BODY
+        # rubocop:enable Layout/HeredocIndentation
+      end
+      let(:mocked_message) do
+        {
+          message_id: "44c44782-fee1-6784-d614-43b73c0bda8d",
+          receipt_handle: "2312dasdas1231221312321adsads",
+          body: body_json
+        }
+      end
+
+      it "should consume a message " do
+        mocked_poller = Aws::SQS::QueuePoller.new("http://localhost:4566/000000000000/my_queue",
+                                                  { skip_delete: true })
+        mocked_poller.before_request { |stats| throw :stop_polling if stats.received_message_count > 0 }
+
+        mocked_element = Aws::SQS::Types::Message.new(mocked_message)
+        mocked_list = Aws::Xml::DefaultList.new
+        mocked_list.append(mocked_element)
+        mocked_poller.client.stub_responses(:receive_message, messages: mocked_list)
+
+        worker = described_class.new
+        worker.instance_variable_set(:@poller, mocked_poller)
+
+        worker.poller do |message, metadata|
+          expect(message).to eq expected_message_result
+          expect(metadata).to eq expected_metadata_result
+        end
+      end
+    end
+
+    context "Raw Message Delivery enabled" do
+      let(:mocked_message) do
+        {
+          message_id: "44c44782-fee1-6784-d614-43b73c0bda8d",
+          receipt_handle: "2312dasdas1231221312321adsads",
+          body: "{\"default\":{\"foo\":\"bar\"}}",
+          message_attributes: {
+            "context" => {
+              data_type: "String",
+              string_value: context
+            },
+            "correlationId" => {
+              data_type: "String",
+              string_value: correlationId
+            },
+            "eventId" => {
+              data_type: "String",
+              string_value: eventId
+            }
+          }
+        }
+      end
+
+      it "should consume a message " do
+        mocked_poller = Aws::SQS::QueuePoller.new("http://localhost:4566/000000000000/my_queue",
+                                                  { skip_delete: true })
+        mocked_poller.before_request { |stats| throw :stop_polling if stats.received_message_count > 0 }
+
+        mocked_element = Aws::SQS::Types::Message.new(mocked_message)
+        mocked_list = Aws::Xml::DefaultList.new
+        mocked_list.append(mocked_element)
+        mocked_poller.client.stub_responses(:receive_message, messages: mocked_list)
+
+        worker = described_class.new
+        worker.instance_variable_set(:@poller, mocked_poller)
```

Comment:
These lines to setup the test are duplicated with lines 104-114. Do you think we can extract it into a `before` block?

Reply:
Done!